python/mlcroissant/mlcroissant/_src/structure_graph/nodes/metadata.py

```diff
@@ -130,6 +144,12 @@ def to_json(self) -> Json:
         date_published = (
             self.date_published.isoformat() if self.date_published else None
         )
+        data_collection_timeframe_start = (
+            self.data_collection_timeframe_start.isoformat() if self.data_collection_timeframe_start else None
```

Comment:
You should also update methods `Metadata.from_jsonld`/`Metadata.to_json` that respectively serialize/deserialize the JSON-LD

Reply:
As you pointed out earlier, the way to test the deserialization of the JSON-LD would be to update some dataset examples. However, example datasets are not properly loading, as I pointed out here: 
https://github.com/mlcommons/croissant/issues/478